test/hotspot/jtreg/vmTestbase/nsk/aod/VirtualMachine/VirtualMachine03/VirtualMachine03.java

```diff
@@ -63,23 +63,21 @@ public void doTestActions(String targetVMId) throws Throwable {
         String currentVMId = getCurrentVMId();
 
         VirtualMachine vm1 = VirtualMachine.attach(currentVMId);
-
-        VirtualMachine vm2 = VirtualMachine.attach(targetVMId);
-
         try {
-            TestUtils.assertEquals(vm1.id(), currentVMId, "vm.id() returns unexpected value: " + vm1.id());
-
-            TestUtils.assertEquals(vm2.id(), targetVMId, "vm.id() returns unexpected value: " + vm2.id());
-
-            TestUtils.assertTrue(!vm1.equals(vm2), vm1 + ".equals(" + vm2 + ") returns 'true'");
-
-            checkVM(vm1);
-
-            checkVM(vm2);
+            VirtualMachine vm2 = VirtualMachine.attach(targetVMId);
+            try {
+                TestUtils.assertEquals(vm1.id(), currentVMId, "vm.id() returns unexpected value: " + vm1.id());
+                TestUtils.assertEquals(vm2.id(), targetVMId, "vm.id() returns unexpected value: " + vm2.id());
+
+                TestUtils.assertTrue(!vm1.equals(vm2), vm1 + ".equals(" + vm2 + ") returns 'true'");
+
+                checkVM(vm1);
+                checkVM(vm2);
+            } finally {
+                vm2.detach();
+            }
```

Comment:
Is nesting needed here because `vm1`/`vm2` can be `null` at `finally`? Then it is cleaner IMO to just do:

```
} finally {
    if (vm1 != null) vm1.detach();
    if (vm2 != null) vm2.detach();
}
```

Reply:
nested try-finally isn't semantically equivalent with the snippet you suggested, `VirtualMachine::detach` can throw IOE, so I'd prefer to two finally blocks to guarantee that we at least to detach from both vm-s.

Reply:
Okay then.